tools/Nethermind.Tools.Kute/MetricsConsumer/JsonMetricsConsumer.cs

```diff
@@ -1,52 +1,20 @@
 // SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-using System.Text.Json;
-using System.Text.Json.Serialization;
+using App.Metrics.Formatters.Json;
 
 namespace Nethermind.Tools.Kute.MetricsConsumer;
 
 public class JsonMetricsConsumer : IMetricsConsumer
 {
-    public void ConsumeMetrics(Metrics metrics)
+    public async Task ConsumeMetrics(Metrics metrics)
     {
-        var metricsObject = new
-        {
-            metrics.TotalRunningTime,
-            Messages = new
-            {
-                metrics.Failed,
-                Succeeded = new
-                {
-                    metrics.Responses,
-                    Requests = new
-                    {
-                        metrics.Batches,
-                        Singles = new
-                        {
-                            Ignored = metrics.IgnoredRequests, Processed = metrics.ProcessedRequests
-                        }
-                    }
-                }
-            }
-        };
-
-        string json = JsonSerializer.Serialize(
-            metricsObject,
-            new JsonSerializerOptions
-            {
-                PropertyNamingPolicy = JsonNamingPolicy.CamelCase, Converters = { new TimeSpanInMsConverter() }
-            }
-        );
-        Console.WriteLine(json);
-    }
+        var snapshot = metrics.Snapshot;
+        var formatter = new MetricsJsonOutputFormatter();
```

Comment:
potentially you could create just one and store it in a field.

Reply:
Done, formatter is now a class field.